commons/network/src/main/java/in/projecteka/jataayu/network/utils/RetrofitCallback.kt

```diff
@@ -0,0 +1,53 @@
+package `in`.projecteka.jataayu.network.utils
+
+import `in`.projecteka.jataayu.network.model.Error
+import `in`.projecteka.jataayu.network.model.ErrorResponse
+import androidx.lifecycle.MutableLiveData
+import com.google.gson.Gson
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import java.net.HttpURLConnection.HTTP_FORBIDDEN
+import java.net.HttpURLConnection.HTTP_UNAUTHORIZED
+
+private const val DEFAULT_ERROR_CODE = -1
+private const val ERROR_CODE_UNAUTHORIZED = 1017
+
+abstract class RetrofitCallback<T> : Callback<T?> {
+    private var responseCallback: ResponseCallback? = null
+
+    constructor()
+
+    constructor(responseCallback: ResponseCallback?) {
+        this.responseCallback = responseCallback
+    }
+
+    protected abstract fun observableLiveData(): MutableLiveData<T?>
```

Comment:
can instead make this a part of constructor, since that is the contract anyway

Reply:
No, there are cases where I don't want a callback/progress bar. e.g. Search. It was kept optional for that only.